onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -512,208 +512,348 @@ Status EmbedLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
     }
     Node& layer_norm_add_node = *graph.GetNode(edges[0]->GetNode().Index());
 
-    // Trace back to find the Gather for segment embedding.
-    std::vector<graph_utils::EdgeEndToMatch> segment_embedding_path{
-        {0, 1, "Gather", {1, 11}, kOnnxDomain}};
-    if (!graph_utils::FindPath(layer_norm_add_node, true, segment_embedding_path, edges, logger)) {
-      continue;
-    }
-    Node& segment_gather_node = *graph.GetNode(edges[0]->GetNode().Index());
-    if (!optimizer_utils::CheckOutputEdges(graph, segment_gather_node, 1)) {
-      continue;
-    }
-    // The first input of segment_gather_node must be 2d.
-    NodeArg* segment_embedding = segment_gather_node.MutableInputDefs()[0];
-    auto sg_shape = segment_embedding->Shape();
-    if (sg_shape == nullptr || sg_shape->dim_size() != 2 ||
-        !utils::HasDimValue(sg_shape->dim()[1]) ||
-        sg_shape->dim()[1].dim_value() <= 0) {
-      continue;
+    // Check if it's distil-bert
+    int add_count = 0;
+    int gather_count = 0;
+    for (auto it = layer_norm_add_node.InputNodesBegin(); it != layer_norm_add_node.InputNodesEnd(); ++it) {
```

Comment:
Instead of counting node type, why not check that the add node has two input nodes, and both are Gather.